### PR TITLE
Run a production server in CI

### PR DESCRIPTION
**Pull request by Technici4n**

### CODE REVIEW

File: .github/workflows/build-prs.yml [hunk 1]
```diff
@@ -51,3 +51,18 @@ jobs:
...
+          # Find installer jar
+          INSTALLER_JAR=$(find projects/neoforge/build/libs -type f -iname "neoforge-*-installer.jar" | tail -n 1)
+          # Run installer
+          java -jar $INSTALLER_JAR --install-server prod-server/
+
+      - name: Run production server
+        run: |
+          cd prod-server/
+          # Accept EULA
+          echo "eula=true" >> eula.txt
```
Comment (Matyrobbrt) [inline .github/workflows/build-prs.yml:66]: This isn't nedded anymore as `IS_RUNNING_IN_IDE` is `true`.

**Review by Matyrobbrt — CHANGES_REQUESTED** [review]
Should also note that running the server like this will delay PR publishing by quite a bit. Maybe it would make more sense to have it as a separate action that runs post build-prs.

File: .github/workflows/build-prs.yml (hunk 1 above)
Comment (Technici4n) [reply .github/workflows/build-prs.yml:66]: not with the installer, right?

Comment (Matyrobbrt) [reply .github/workflows/build-prs.yml:66]: Oh right, ignore me.

**Comment by Technici4n** [comment]
Note: add a timeout.

**Review by marchermans — CHANGES_REQUESTED** [review]
Move this to a seperate workflow and we have a deal I would say.

File: src/main/java/net/neoforged/neoforge/common/NeoForgeMod.java [hunk 2]
```diff
@@ -630,6 +633,20 @@ public NeoForgeMod(IEventBus modEventBus, Dist dist, ModContainer container) {
...
+            NeoForge.EVENT_BUS.addListener(TickEvent.ServerTickEvent.class, event -> {
+                if (event.phase == TickEvent.Phase.END) {
+                    if (serverTicks.incrementAndGet() == 50) {
+                        MixinEnvironment.getCurrentEnvironment().audit();
+                        event.getServer().halt(false);
+                        LOGGER.info("Server auto-test successful");
+                    }
+                }
+            });
+        }
```
Comment (shartte) [inline src/main/java/net/neoforged/neoforge/common/NeoForgeMod.java:649]: I'd probably not put this in the actual NeoForge mod and instead make a separate mod hm.
Although I suppose you could call it "selfTest"

Comment (Technici4n) [reply src/main/java/net/neoforged/neoforge/common/NeoForgeMod.java:649]: Another mod is a bit of cruft for nothing IMO. We may want to document that this is purely internal and not intended for modders to enable.

Comment (shartte) [reply src/main/java/net/neoforged/neoforge/common/NeoForgeMod.java:649]: What you get is that it actually validates mod-loading works (from `mods/`)

**Pull request closed without merging**